[MODULE2-PART-1] Move project from JavaScript to TypeScript

snikolenko-flo opened the pull request.

snikolenko-flo committed:
chore: install environment

install environment

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change gallery to typescript

change gallery to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change handler to typescript

change handler to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change gallery manager to typescript

change gallery manager to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change gallery service to typescript

change gallery service to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change gallery api to typescript

change gallery api to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change gallery render to typescript

change gallery render to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change gallery.url to typescript

change gallery.url to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change login handler to typescript

change login handler to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change login.js to typescript

change login.js to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change login manager to typescript

change login manager to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change login service to typescript

change login service to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change url service to typescript

change url service to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change token handler to typescript

change token handler to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change token manager to typescript

change token manager to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change token service to typescript

change token service to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change token setting to typescript

change token setting to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change token checking to typescript

change token checking to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor: change constants file to typescript

change constants file to typescript

MODULE-2-PART-1-typescript

snikolenko-flo committed:
fix(gallery): add page word to login after redirect

fix the issue: “page” word is not appeared when we are redirected to login page from gallery page

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor(gallery): change abbreviations according to Google Style Guide

change abbreviations according to Google Style Guide

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor(gallery): add feature description to readme

add feature description to readme

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor(gallery): add building information to readme

add building information to readme

MODULE-2-PART-1-typescript

arevta-flo commented on package.json:
```diff
@@ -0,0 +1,33 @@
+  "version": "1.0.0",
+  "main": "webpack.config.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
please, add scripts for build and lint. set up eslint and prettier using google config.

arevta-flo commented on src/gallery/gallery.service.ts:
```diff
@@ -1,6 +1,6 @@
 export class GalleryService {
-    wrapUrlsInHtml(urlsList) {
-        let images = '';
+    wrapUrlsInHtml(urlsList: URL[]) {
```
please, add return type in all functions when it is absent

arevta-flo commented on src/token/token.checking.ts:
```diff
@@ -2,24 +2,27 @@ import { UrlManipulationService } from "../services/url-manipulation.service.js"
-    #redirectToLogin() {
-        const pageNumber = this.urlService.getPageNumberFromUrl();
-        window.location.href = `../login/login.html?page=${pageNumber}`;
+    private redirectToLogin() {
```
this method can be moved into service

arevta-flo commented on src/login/handler.ts:
```diff
@@ -2,21 +2,21 @@ import { LoginManager } from "./login.manager.js";
     return () => {
-        const emailError = document.getElementById('emailError');
+        const emailError = document.getElementById('emailError') as HTMLFormElement;
         manager.checkEmail(formElement, emailError);
```
you can pass only email here

arevta-flo commented on src/login/handler.ts:
```diff
@@ -2,21 +2,21 @@ import { LoginManager } from "./login.manager.js";
 
-        const passwordError = document.getElementById('passwordError');
+        const passwordError = document.getElementById('passwordError') as HTMLFormElement;
         manager.checkPassword(formElement, passwordError);
```
you can pass only password here

arevta-flo commented on src/token/token.setting.ts:
```diff
@@ -0,0 +1,14 @@
+    }
+
+    setExpireTime() {
+        const oneMinuteInMs: number = 60000;
```
you can make this more flexible by moving minutes value into env variables

snikolenko-flo committed:
refactor(gallery): pass user credentials as strings

pass email and password as strings, not as html forms as now

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor(gallery): create a constant for minutes

create a constant for minutes

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor(gallery): move login redirect to services

move login redirect to services

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor(gallery): add return type to all functions

add return type to all functions

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor(gallery): setup eslint and prettier

setup eslint and prettier, check code and make changes accordingly

MODULE-2-PART-1-typescript

arevta-flo commented on src/token/token.setting.ts:
```diff
@@ -1,12 +1,12 @@
-        const oneMinuteInMs: number = 60000;
-        const tenMinutes: number = oneMinuteInMs * 10;
-
+        const tenMinutes: number = ONE_MINUTE * 10;
```
what if you would like to change expiration time to another value? It's okay to keep ms in one minute in the class property. It's better to keep concrete value of minutes in env variables to be able to change this value if it needed without changing the code

arevta-flo commented on .eslintrc.json:
```diff
@@ -0,0 +1,13 @@
+{
```
I cannot apply code style rules from the configuration file because several eslint plugin packages are absent

snikolenko-flo committed:
refactor(gallery): create a constant for 10 minutes

create a constant for 10 minutes

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor(gallery): create a constant for token expiration time

create a constant for token expiration time

MODULE-2-PART-1-typescript

snikolenko-flo committed:
refactor(gallery): setup eslint to show errors

setup eslint to show errors in IDE

MODULE-2-PART-1-typescript

arevta-flo commented on src/token/token.setting.ts:
```diff
@@ -1,12 +1,19 @@
 
   setExpireTime(): void {
-    const tokenExpireTime: number = Date.now() + TEN_MINUTES;
+    const tenMinutes = EXPIRATION_TIME * this.oneMinuteInMs;
```
if you change expiration time value, this const name won't be actual)

arevta-flo commented on src/token/token.setting.ts:
```diff
@@ -1,12 +1,19 @@
   setExpireTime(): void {
-    const tokenExpireTime: number = Date.now() + TEN_MINUTES;
+    const tenMinutes = EXPIRATION_TIME * this.oneMinuteInMs;
+    const tokenExpireTime: number = Date.now() + tenMinutes;
```
you can multiply ms and expiration time right here. it won't make the code more difficult to read

snikolenko-flo committed:
refactor(gallery): remove ten minutes variable

remove ten minutes variable

MODULE-2-PART-1-typescript
